app.py

```diff
@@ -268,130 +241,185 @@ def update_button_visibility(running_dwave: bool, running_classical: bool) -> tu
-    Output({"type": "sort-button", "index": MATCH}, "children"),
+    Output({"type": "gantt-chart-visible-wrapper", "index": MATCH}, "children"),
+    Output({"type": "gantt-chart-hidden-wrapper", "index": MATCH}, "children"),
+    Output({"type": "gantt-heading-button", "index": MATCH}, "children"),
     inputs=[
-        Input({"type": "sort-button", "index": MATCH}, "n_clicks"),
-        State({"type": "sort-button", "index": MATCH}, "children"),
+        Input({"type": "gantt-heading-button", "index": MATCH}, "n_clicks"),
+        State({"type": "gantt-heading-button", "index": MATCH}, "children"),
+        State({"type": "gantt-chart-visible-wrapper", "index": MATCH}, "children"),
+        State({"type": "gantt-chart-hidden-wrapper", "index": MATCH}, "children"),
     ],
     prevent_initial_call=True,
 )
-def switch_gantt_chart(new_click: int, sort_button_text: str) -> tuple[str, str, str]:
+def switch_gantt_chart(
+    new_click: int, sort_button_text: str, visibleChart: list, hiddenChart: list
+) -> tuple[str, str, str]:
     """Switch between the results plot sorted by job or by start time.
 
     Args:
         new_click (int): The number of times the sort button has been clicked.
-        sort_button_text: The text in the sort button (indicating how to sort the plot).
+        sort_button_text (str): The text of the sort button (indicating how to sort the plot).
+        visibleChart (list): The children of the currently visible graph.
+        hiddenChart (list): The children of the currently hidden graph.
 
     Return:
-        str: The results class name sorted by job (whether hidden or displayed).
-        str: The results class name sorted by start time (whether hidden or displayed).
+        list: The new graph that should be visible.
+        list: The new graph that should be hidden.
         str: The new text of the sort button.
     """
-    if sort_button_text == "Sort by start time":
-        return "display-none", "gantt-div", "Sort by job"
-    return "gantt-div", "display-none", "Sort by start time"
+    if ctx.triggered_id["index"] == 0:
+        button_text = "Show Conflicts" if sort_button_text == "Hide Conflicts" else "Hide Conflicts"
+    else:
+        button_text = (
+            "Sort by job" if sort_button_text == "Sort by start time" else "Sort by start time"
+        )
+    return hiddenChart, visibleChart, button_text
+
+
+def get_table_rows(
+    scenario: str, solver: str, model_data: JobShopData, time_limit: int, wall_clock_time: float
+) -> tuple[dict]:
+    """Returns tuple of dicts where each dict represents a row in the problem details table."""
+    return (
+        {"Scenario": scenario, "Solver": solver},
+        {"Number of Jobs": model_data.get_job_count(), "Solver Time Limit": f"{time_limit}s"},
+        {
+            "Number of Operations": model_data.get_resource_count(),
+            "Wall Clock Time": f"{round(wall_clock_time, 2)}s",
+        },
```

Comment:
Why not just put this inside the `generate_problem_details_table` function instead?

Reply:
updated: https://github.com/dwave-examples/flow-shop-scheduling/pull/5/commits/6c9e8ca64844a1f71962ed15d87350bde9b8ee47